errors/deleting-query-params-in-middlewares.md

```diff
@@ -0,0 +1,40 @@
+# Deleting Query Parameters In Middlewares
+
+#### Why This Error Occurred
+
+In previous versions of Next.js, we were merging query parameters with the incoming request for rewrites happening in middlewares, to match the behavior of static rewrites declared in the config. This forced Next.js users to use empty query parameters values to delete keys.
+
+We are changing this behavior to allow extra flexibility and a more streamlined experience for users. So from now on, query parameters will not be merged and thus the warning.
+
+```typescript
+import type { NextRequest } from 'next/server'
+import { NextResponse } from 'next/server'
+
+export default function middleware(request: NextRequest) {
+  const nextUrl = request.nextUrl
+  nextUrl.searchParams.delete('key') // this is possible
+  return NextResponse.rewrite(nextUrl)
+}
+```
+
+#### Possible Ways to Fix It
+
+If you are relying on the old behavior, please add the query parameters manually to the rewritten URL:
+
+```typescript
+import type { NextRequest } from 'next/server'
+import { NextResponse } from 'next/server'
+
+export default function middleware(request: NextRequest) {
+  const url = new URL('/dest', request.url)
```

Comment:
I don't like much this practice because it is `URL` and not `NextURL`. Maybe the example is more in line with others if we do:

```typescript
const url = request.nextUrl.clone();
url.pathname = '/dest';
return NextResponse.rewrite(url);
```

Reply:
I wanted to show how to forcefully take the search params. But yeah I'd rather go for the nextUrl as it is easier to follow 👍

Reply:
Should I also mention "reusing `request.nextUrl` would do that automatically for you" or something?